### PR TITLE
resource/aws_route53_resolver_firewall_rule: Add new resource

### PR DESCRIPTION
**Pull request by gazoakley**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #18520

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSRoute53ResolverFirewallRule_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRoute53ResolverFirewallRule_ -timeout 180m
=== RUN   TestAccAWSRoute53ResolverFirewallRule_basic
=== PAUSE TestAccAWSRoute53ResolverFirewallRule_basic
=== RUN   TestAccAWSRoute53ResolverFirewallRule_block
=== PAUSE TestAccAWSRoute53ResolverFirewallRule_block
=== RUN   TestAccAWSRoute53ResolverFirewallRule_blockOverride
=== PAUSE TestAccAWSRoute53ResolverFirewallRule_blockOverride
=== RUN   TestAccAWSRoute53ResolverFirewallRule_disappears
=== PAUSE TestAccAWSRoute53ResolverFirewallRule_disappears
=== CONT  TestAccAWSRoute53ResolverFirewallRule_basic
=== CONT  TestAccAWSRoute53ResolverFirewallRule_disappears
=== CONT  TestAccAWSRoute53ResolverFirewallRule_block
=== CONT  TestAccAWSRoute53ResolverFirewallRule_blockOverride
--- PASS: TestAccAWSRoute53ResolverFirewallRule_disappears (96.17s)
--- PASS: TestAccAWSRoute53ResolverFirewallRule_block (98.79s)
--- PASS: TestAccAWSRoute53ResolverFirewallRule_blockOverride (99.04s)
--- PASS: TestAccAWSRoute53ResolverFirewallRule_basic (99.51s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	103.330s
```

### CODE REVIEW

**Comment by gazoakley** [comment]
Based on #18558 - will rebase if necessary after review

File: aws/resource_aws_route53_resolver_firewall_rule.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+		Delete: resourceAwsRoute53ResolverFirewallRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```
Comment (ewbankkit) [inline aws/resource_aws_route53_resolver_firewall_rule.go:30]: No need for explicit `id` attribute.

**Comment by gazoakley** [comment]
Hi @ewbankkit - was about to make those same changes myself 😄 . The sweeper also needs fixing - I'll take a look at that shortly

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRoute53ResolverFirewallRule_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRoute53ResolverFirewallRule_ -timeout 180m
=== RUN   TestAccAWSRoute53ResolverFirewallRule_basic
=== PAUSE TestAccAWSRoute53ResolverFirewallRule_basic
=== RUN   TestAccAWSRoute53ResolverFirewallRule_block
=== PAUSE TestAccAWSRoute53ResolverFirewallRule_block
=== RUN   TestAccAWSRoute53ResolverFirewallRule_blockOverride
=== PAUSE TestAccAWSRoute53ResolverFirewallRule_blockOverride
=== RUN   TestAccAWSRoute53ResolverFirewallRule_disappears
=== PAUSE TestAccAWSRoute53ResolverFirewallRule_disappears
=== CONT  TestAccAWSRoute53ResolverFirewallRule_basic
=== CONT  TestAccAWSRoute53ResolverFirewallRule_disappears
=== CONT  TestAccAWSRoute53ResolverFirewallRule_blockOverride
=== CONT  TestAccAWSRoute53ResolverFirewallRule_block
--- PASS: TestAccAWSRoute53ResolverFirewallRule_disappears (76.91s)
--- PASS: TestAccAWSRoute53ResolverFirewallRule_block (78.08s)
--- PASS: TestAccAWSRoute53ResolverFirewallRule_basic (79.02s)
--- PASS: TestAccAWSRoute53ResolverFirewallRule_blockOverride (79.29s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	79.384s
```

**Pull request merged**